qcodes/instrument_drivers/rohde_schwarz/SGS100A.py

```diff
@@ -27,110 +36,220 @@ class RohdeSchwarz_SGS100A(VisaInstrument):
-                           get_cmd=':OUTP:STAT?',
-                           set_cmd=self.set_status,
-                           get_parser=self.parse_on_off,
-                           vals=vals.Strings())
-        self.add_parameter('IQ_state',
-                           get_cmd=':IQ:STAT?',
-                           set_cmd=self.set_IQ_state,
-                           get_parser=self.parse_on_off,
-                           vals=vals.Strings())
-        self.add_parameter('pulsemod_state',
-                           get_cmd=':SOUR:PULM:STAT?',
-                           set_cmd=self.set_pulsemod_state,
-                           get_parser=self.parse_on_off,
-                           vals=vals.Strings())
-        self.add_parameter('pulsemod_source',
-                           get_cmd='SOUR:PULM:SOUR?',
-                           set_cmd=self.set_pulsemod_source,
-                           vals=vals.Strings())
-        self.add_parameter('ref_osc_source',
-                           label='Reference oscillator source',
-                           get_cmd='SOUR:ROSC:SOUR?',
-                           set_cmd='SOUR:ROSC:SOUR {}',
-                           vals=vals.Enum('INT', 'EXT'))
-        # Frequency mw_source outputs when used as a reference
-        self.add_parameter('ref_osc_output_freq',
-                           label='Reference oscillator output frequency',
-                           get_cmd='SOUR:ROSC:OUTP:FREQ?',
-                           set_cmd='SOUR:ROSC:OUTP:FREQ {}',
-                           vals=vals.Enum('10MHz', '100MHz', '1000MHz'))
-        # Frequency of the external reference mw_source uses
-        self.add_parameter('ref_osc_external_freq',
-                           label='Reference oscillator external frequency',
-                           get_cmd='SOUR:ROSC:EXT:FREQ?',
-                           set_cmd='SOUR:ROSC:EXT:FREQ {}',
-                           vals=vals.Enum('10MHz', '100MHz', '1000MHz'))
-
-        self.add_function('reset', call_cmd='*RST')
-        self.add_function('run_self_tests', call_cmd='*TST?')
+    def __init__(self, name, address, max_frequency=12.75e9, **kwargs):
+        super().__init__(name, address, terminator="\n", **kwargs)
+
+        self.add_parameter(
+            name="frequency",
+            label="Frequency",
+            unit="Hz",
+            get_cmd="SOUR:FREQ?",
+            set_cmd="SOUR:FREQ {:.2f}",
+            get_parser=float,
+            vals=vals.Numbers(1e6, max_frequency)
+        )
+        self.add_parameter(
+            name="phase",
+            label="Phase",
+            unit="deg",
+            get_cmd="SOUR:PHAS?",
+            set_cmd="SOUR:PHAS {:.2f}",
+            get_parser=float,
+            vals=vals.Numbers(0, 360)
+        )
+        self.add_parameter(
+            name="power",
+            label="Power",
+            unit="dBm",
+            get_cmd="SOUR:POW?",
+            set_cmd="SOUR:POW {:.2f}",
```

Comment:
From the [manual](https://www.rohde-schwarz.com/webhelp/sgs100a_html_usermanual_en/SGS100A_HTML_UserManual_en.htm) it looks like the right command is:
`':SOURce:POWer:POWer {:.2f}'`

Does it work without the second power?

Reply:
Haha interesting, it seems to work, I'll check

Reply:
Yes apparently it works